lib/plugins/aws/provider/awsProvider.js

```diff
@@ -56,13 +57,20 @@ const impl = {
    */
   addProfileCredentials: (credentials, profile) => {
     if (profile) {
-      const profileCredentials = new AWS.SharedIniFileCredentials({ profile });
+      const profileCredentials = impl.getIniProfileCredentials(profile);
       if (Object.keys(profileCredentials).length) {
         credentials.profile = profile; // eslint-disable-line no-param-reassign
       }
       impl.addCredentials(credentials, profileCredentials);
     }
   },
+  getIniProfileCredentials: (profile) => {
```

Comment:
doh! thanks for catching that. running the entire test suite also turned up the need to do a better job of sandboxing envars.